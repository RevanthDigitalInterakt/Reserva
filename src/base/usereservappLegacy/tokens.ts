export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
