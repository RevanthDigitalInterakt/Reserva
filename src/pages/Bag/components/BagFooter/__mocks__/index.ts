import { profileQuery } from '../../../../../graphql/profile/profileQuery';

export const bagInfos = {
  totalBagItemsPrice: 100,
  totalBagItems: 20,
  totalBagDiscountPrice: 40,
  totalBagDeliveryPrice: 60,
};

export const currentOrderForm: any = {
  items: [
    {
      name: 'Produto A',
      skuName: 'test',
      price: 50.0,
      quantity: 2,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
    {
      name: 'Produto B',
      skuName: 'test',
      price: 20.0,
      quantity: 1,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
  ],
  orderFormId: '1',
};

export const installmentInfo = {
  installmentPrice: 110,
  installmentsNumber: 40,
  totalPrice: 1000,
};

export const apolloMocks = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          addresses: [[{}]],
          customFields: [[{}], [{}], [{}]],
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: 'Tester',
          lastName: 'Silva',
          fullName: 'teste da Silva',
          email: 'nogueirahy@gmail.com',
          document: '11111111111',
          birthDate: '1980-01-01T00:00:00.000Z',
          gender: 'male',
          homePhone: '+5511991111111',
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];

export const apolloMocksWithoutDataUser = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: null,
          lastName: null,
          fullName: null,
          email: 'nogueirahy@gmail.com',
          document: null,
          birthDate: null,
          gender: 'male',
          homePhone: null,
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];
