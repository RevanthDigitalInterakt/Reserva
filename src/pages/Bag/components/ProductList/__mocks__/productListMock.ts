import type { OrderformOutput } from '../../../../../base/graphql/generated';

export const orderFormMock = {
  orderFormId: '85b805c4c8c04d99b9cc0bf61d1c58e1',
  appTotalizers: {
    items: 828,
    discount: -79.71,
    delivery: 0,
    total: 748.29,
    __typename: 'OrderformAppTotalizersOutput',
  },
  items: [
    {
      productTitle: 'CAMISA ML SAMOA BRANCO',
      itemColor: 'BRANCO',
      itemSize: 'P',
      isGift: false,
      isGiftable: true,
      imageSource: 'https://lojausereserva.vteximg.com.br/arquivos/ids/6413946/0054128014_01.jpg?v=637692018478770000',
      key: '90368-39900-19929-19929-19929-1-1-camisa-ml-samoa-branco-p-',
      isAssinaturaSimples: false,
      priceWithDiscount: 199.29,
      discountPercent: 50,
      discountApi: 0,
      showFirstPurchaseDiscountMessage: null,
      showTotalDiscountFirstPurchaseValue: null,
      price: 27900,
      productId: '3239',
      id: '90368',
      listPrice: 39900,
      giftOfferingId: '90310',
      seller: '1',
      skuName: 'BRANCO - P',
      uniqueId: '52D9B57139FA4292909748AE8ED11774',
      isAddedAsGift: false,
      name: 'CAMISA ML SAMOA BRANCO - P',
      quantity: 1,
      disableCounter: false,
      sellingPrice: 19929,
      __typename: 'OrderformItemOutput',
    },
    {
      productTitle: 'TENIS RSV NEO RE BRANCO/VERDE MILITAR',
      itemColor: 'BRANCO/VERDE MILITAR',
      itemSize: '41',
      isGift: false,
      isGiftable: true,
      imageSource: 'https://lojausereserva.vteximg.com.br/arquivos/ids/7575487/0075733901_01.jpg?v=638145040244900000',
      key: '426190-54900-54900-54900-54900-lojausereservago-1-tenis-rsv-neo-re-branco-verde-militar-41-',
      isAssinaturaSimples: false,
      priceWithDiscount: 549,
      discountPercent: 0,
      discountApi: 0,
      showFirstPurchaseDiscountMessage: null,
      showTotalDiscountFirstPurchaseValue: null,
      price: 54900,
      productId: '40572',
      id: '426190',
      listPrice: 54900,
      giftOfferingId: '27623',
      seller: 'lojausereservago',
      skuName: 'BRANCO/VERDE MILITAR - 41',
      uniqueId: '49A868C5C6974936AD7BA870A830A7A8',
      isAddedAsGift: false,
      name: 'TENIS RSV NEO RE BRANCO/VERDE MILITAR - 41',
      quantity: 1,
      disableCounter: false,
      sellingPrice: 54900,
      __typename: 'OrderformItemOutput',
    },
  ],
  selectableGift: null,
  allItemsQuantity: 2,
  installmentInfo: {
    installmentsNumber: 12,
    installmentPrice: 62.35,
    totalPrice: 748.29,
    __typename: 'OrderformInstallmentInfoOutput',
  },
  marketingData: null,
  __typename: 'OrderformOutput',
} as OrderformOutput;
