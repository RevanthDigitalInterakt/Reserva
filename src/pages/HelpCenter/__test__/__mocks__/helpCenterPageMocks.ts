import type { DocumentNode } from 'graphql';
import { HelpCenterCollectionDocument, type HelpCenterCollectionQuery } from '../../../../base/graphql/generated';
import { mockHelpCenterCollectionData } from '../../../../zustand/useHelpCenterStore/mocks/helpCenterCollection.mock';

interface IApolloMock<T> {
  request: {
    query: DocumentNode,
    variables: object,
  },
  result: {
    data: T
  }
}

export const helpCenterPageMocks: Array<IApolloMock<HelpCenterCollectionQuery>> = [
  {
    request: {
      query: HelpCenterCollectionDocument,
      variables: {},
    },
    result: {
      data: {
        helpCenterCollection: {
          ...mockHelpCenterCollectionData,
        },
        __typename: 'Query',
      },
    },
  },
];
