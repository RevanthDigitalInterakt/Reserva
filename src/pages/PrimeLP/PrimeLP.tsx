import type { StackScreenProps } from '@react-navigation/stack';
import React, {
  useCallback, useMemo, useState,
} from 'react';
import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { useLandingPagePrimeQuery } from '../../base/graphql/generated';
import { ModalBag } from '../../components/ModalBag/ModalBag';
import { useApolloFetchPolicyStore } from '../../zustand/useApolloFetchPolicyStore';
import testProps from '../../utils/testProps';
import { ModalWelcomePrime } from '../../components/ModalWelcomePrime';
import { usePrimeInfo } from '../../hooks/usePrimeInfo';
import { TopBarDefaultBackButton } from '../../modules/Menu/components/TopBarDefaultBackButton';
import type { RootStackParamList } from '../../routes/StackNavigator';
import { usePrimeStore } from '../../zustand/usePrimeStore/usePrimeStore';
import PrimeBenefits from './components/PrimeBenefits';
import PrimeFAQ from './components/PrimeFAQ/PrimeFAQ';
import PrimeHero from './components/PrimeHero';
import PrimeIntro from './components/PrimeIntro';
import PrimeSubscribe from './components/PrimeSubscribe';
import { Box } from '../../components/Box/Box';
import { ExceptionProvider } from '../../base/providers/ExceptionProvider';

type IPrimeLP = StackScreenProps<RootStackParamList, 'PrimeLP'>;

function PrimeLP({ navigation }: IPrimeLP) {
  const { onAddPrimeToCart, isPrime } = usePrimeInfo();
  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);
  const [loadingAddCartPrime, setLoadingAddCartPrime] = useState(false);

  const {
    animationBag,
    handleClickContinue,
    isVisibleModalWelcome,
    changeStateAnimationBag,
    changeStateIsVisibleModalWelcome,
  } = usePrimeStore([
    'animationBag',
    'handleClickContinue',
    'isVisibleModalWelcome',
    'changeStateAnimationBag',
    'changeStateIsVisibleModalWelcome',
  ]);

  const { data: rawData, loading } = useLandingPagePrimeQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('landingPagePrime'),
  });

  const data = useMemo(() => rawData?.landingPagePrime, [rawData?.landingPagePrime]);

  const handleOnModalHideSignIn = useCallback(() => {
    changeStateAnimationBag(true);
  }, [changeStateAnimationBag]);

  const handleOnModalHide = useCallback(() => {
    if (isPrime) {
      setTimeout(() => {
        changeStateIsVisibleModalWelcome(true);
      }, 500);
    }
  }, [changeStateIsVisibleModalWelcome, isPrime]);

  const onAddToCart = useCallback(async () => {
    try {
      if (!data || loadingAddCartPrime) return;

      if (isPrime) {
        navigation.popToTop();
        return;
      }

      setLoadingAddCartPrime(true);

      await onAddPrimeToCart(false);

      handleOnModalHideSignIn();
    } catch (e) {
      ExceptionProvider.captureException(e);
    } finally {
      setLoadingAddCartPrime(false);
    }
  }, [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);

  const onCloseModalWelcomePrime = useCallback(() => {
    handleClickContinue();

    setTimeout(() => {
      navigation.navigate('Home');
    }, 300);
  }, [handleClickContinue, navigation]);

  return (
    <SafeAreaView style={{ backgroundColor: '#fff' }}>
      <Box bg="white">
        <TopBarDefaultBackButton
          loading={loading || loadingAddCartPrime}
          navigateGoBack
        />

        <ModalBag
          isVisible={animationBag}
          onBackdropPress={() => changeStateAnimationBag(false)}
          onModalHide={handleOnModalHide}
        />

        <ModalWelcomePrime
          isVisible={isVisibleModalWelcome}
          onClose={onCloseModalWelcomePrime}
        />

        {!!(data && !loading) && (
          <ScrollView
            contentContainerStyle={{ paddingBottom: 100 }}
            {...testProps('PrimeLP_page')}
          >
            <PrimeHero data={data} onAddToCart={onAddToCart} />

            <PrimeIntro data={data} />

            <PrimeBenefits data={data} />

            <PrimeSubscribe data={data} onAddToCart={onAddToCart} />

            <PrimeFAQ data={data} />
          </ScrollView>
        )}
      </Box>
    </SafeAreaView>
  );
}

export default PrimeLP;
