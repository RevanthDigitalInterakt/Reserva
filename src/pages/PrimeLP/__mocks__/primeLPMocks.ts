import type { DocumentNode } from 'graphql';
import { LandingPagePrimeDocument, LandingPagePrimeQuery } from '../../../base/graphql/generated';
import { mockPrimeData } from '../../../../__mocks__/PrimeLP.mock';

interface IApolloMock<T> {
  request: {
    query: DocumentNode,
    variables: object,
  },
  result: {
    data: T
  }
}

export const ApolloMockLPPrime: Array<IApolloMock<LandingPagePrimeQuery>> = [
  {
    request: {
      query: LandingPagePrimeDocument,
      variables: {},
    },
    result: {
      data: {
        landingPagePrime: {
          ...mockPrimeData,
        },
        __typename: 'Query',
      },
    },
  },
];
