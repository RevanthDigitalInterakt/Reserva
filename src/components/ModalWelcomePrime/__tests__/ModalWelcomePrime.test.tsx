import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import { ModalWelcomePrime } from '..';
import { ApolloMockLPPrime } from '../../../pages/PrimeLP/__mocks__/primeLPMocks';
import * as useLandingPagePrimeQuery from '../../../base/graphql/generated';
import { mockPrimeData } from '../../../../__mocks__/PrimeLP.mock';
import { theme } from '../../../base/usereservappLegacy/theme';

jest.mock('../../../zustand/useApolloFetchPolicyStore', () => ({
  useApolloFetchPolicyStore: () => ({
    getFetchPolicyPerKey: () => 'network-only',
  }),
}));

jest
  .spyOn(useLandingPagePrimeQuery, 'useLandingPagePrimeQuery')
  .mockReturnValue({
    data: {
      landingPagePrime: {
        ...mockPrimeData,
      },
    },
    loading: false,
  } as any);

const closeModal = jest.fn();

function TestingComponent() {
  return (
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={ApolloMockLPPrime} addTypename={false}>
        <ModalWelcomePrime isVisible onClose={closeModal} />
      </MockedProvider>
    </ThemeProvider>
  );
}

describe('ModalWelcomePrime', () => {
  beforeEach(() => {
    jest.useFakeTimers({ legacyFakeTimers: true });
  });

  it('should render properly', () => {
    render(TestingComponent());

    const titleModal = screen.getByTestId('com.usereserva:id/title');

    expect(titleModal).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    render(TestingComponent());
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call onPressFn when click in continue', () => {
    render(TestingComponent());

    const buttonContinue = screen.getByTestId('com.usereserva:id/modal_welcome_prime_continue');

    fireEvent.press(buttonContinue);

    expect(closeModal).toHaveBeenCalled();
  });
});
