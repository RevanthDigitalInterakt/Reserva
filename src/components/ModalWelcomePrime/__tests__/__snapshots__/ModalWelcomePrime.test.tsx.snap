// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWelcomePrime should match to snapshot 1`] = `
<Modal
  accessibilityLabel="com.usereserva:id/modal_welcome_prime"
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="com.usereserva:id/modal_welcome_prime"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    accessibilityLabel="com.usereserva:id/modal_welcome_prime"
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[MockFunction]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      accessibilityLabel="com.usereserva:id/modal_sign_in_container"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "padding": 32,
          "width": 726,
        }
      }
      testID="com.usereserva:id/modal_sign_in_container"
    >
      <Text
        accessibilityLabel="com.usereserva:id/title"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "ReservaDisplay-Regular",
            "fontSize": 26,
            "marginBottom": 24,
          }
        }
        testID="com.usereserva:id/title"
      >
        Prime
      </Text>
      <View>
        <Text
          style={
            Object {
              "color": "#767676",
              "fontFamily": "ReservaSans-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            }
          }
        >
          Agora você é um cliente
           
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "ReservaDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              }
            }
          >
            Prime
          </Text>
          , para facilitar na sua compra já adicionamos ao seu carrinho a assinatura, ela será concluída junto ao seu pedido, mas lembre-se, ela só será válida para compras acima de
           
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "ReservaSans-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              }
            }
          >
            R$
             
            499
          </Text>
        </Text>
      </View>
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View
            alignItems="stretch"
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "color": "#FFF",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 13,
                  "justifyContent": "center",
                  "marginTop": 24,
                  "minWidth": 130,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 16,
                }
              }
              testID="com.usereserva:id/modal_welcome_prime_continue"
            >
              <View
                alignItems="center"
                flexDirection="row"
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    color="white"
                    fontFamily="nunitoBold"
                    fontSize={13}
                    letterSpacing={1.6}
                    style={
                      Array [
                        Object {
                          "color": "#FFF",
                          "fontFamily": "Nunito-Bold",
                          "fontSize": 13,
                          "letterSpacing": 1.6,
                        },
                      ]
                    }
                  >
                    CONTINUAR COMPRANDO
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
